### ISSUE TITLE
Assets from `node_modules` do not get copied to output folder

### ISSUE THREAD

**Issue by septatrix** [withastro/astro#5599]
### What version of `astro` are you using?

1.6.14

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

When including assets from `node_modules` (e.g. SVG icons) they are not included in the final output generated after building the project.

This applies to using Astro for SSG.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-fyxlgc?file=src/pages/index.astro

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by matthewp** [comment]
You need to import it. See this example: https://stackblitz.com/edit/github-fyxlgc-ocw3rn?file=src%2Fpages%2Findex.astro

**Comment by septatrix** [comment]
In that case my MWE was too small as the full project has an `Icon` component which unifies some options for Devicon/Fontawesome. So the import would have to be dynamic. I got a working prototype for that: https://stackblitz.com/edit/github-fyxlgc-dkpvym?file=src/components/icon.astro

Now, however, Astro (or vite?) pulls in *every* icon even though I only use a dozen unique ones. Is Astro unable to detect which icons are used in this case and has to fall back to including all of them in the final bundle?

Also do you happen to know how this works with the `<Image>`/`<Picture>` components provided by `@astrojs/image`. Does it not work with dynamic imports with variables or would it then also preprocess every icon?

**Comment by matthewp** [comment]
Yeah, that's right, Astro needs to know ahead of time which files are going to be included, it can't wait until after rendering. That wouldn't work for SSR.

You can also put stuff into the `public/` folder and then just use strings to determine what image to render.

**Comment by septatrix** [comment]
Okay, I expected that already, sadly. Anyways: Thank you.

For anyone else stumbling upon this I will leave most information I found about this here. I will probably forget to come back here once I decided what to do but here is what I will look into:

It seems that dynamic imports are handled by a rollup plugin ([`@rollup/plugin-dynamic-import-vars`](https://github.com/rollup/plugins/tree/master/packages/dynamic-import-vars#options)) and I will take a look whether the `include` option could satisfy what I want. Otherwise I will probably see if I can change how my `Icon` component works and pass it the full URL. While this is not as clean, it is probably the best I can get to what I want as I strongly want to avoid including such assets inside the repo itself. I have not yet tried the dynamic imports with `@astrojs/image` but I guess it would also be the most robust solution for that if I used proper full imports.

I think using the [regular expression matching/resolver capabilities](https://github.com/rollup/plugins/tree/master/packages/alias#regular-expression-aliases) of vite's `resolve.alias` might be really helpful here to make the import statements shorter (e.g. something which could transform `@faIcon/circle/regular` and `@devicon/javascript/original` into the full paths) and thus save the most annoying part.
